### PR TITLE
Sending Checkin

### PR DESCRIPTION
**Pull request by whatSocks**
Please review this WIP

### CODE REVIEW

**Comment by whatSocks** [comment]
Mostly done - success needs "share" functionality

File: BlockHunt/CheckinViewController.m [hunk 1]
```diff
@@ -37,19 +39,17 @@ - (void)tabBarController:(UITabBarController *)tabBarController didSelectViewCon
...
+		QRCodeReader *reader = [QRCodeReader readerWithMetadataObjectTypes:@[AVMetadataObjectTypeQRCode]];
+		vc = [QRCodeReaderViewController readerWithCancelButtonTitle:@"Cancel" codeReader:reader startScanningAtLoad:YES showSwitchCameraButton:YES showTorchButton:YES];
+		vc.modalPresentationStyle = UIModalPresentationFormSheet;
 		vc.delegate = self;
-		
 		[vc setCompletionWithBlock:^(NSString *resultAsString) {
 			NSLog(@"Completion with result: %@", resultAsString);
-			
-			//send response to API, see here for example http://api.blockhunt.io/v1/stores/1/qrcode
+			//creating a new checkin :D
```
Comment (tsldh) [inline BlockHunt/CheckinViewController.m]: remove silly comment

File: BlockHunt/CheckinViewController.m [hunk 2]
```diff
@@ -37,19 +39,17 @@ - (void)tabBarController:(UITabBarController *)tabBarController didSelectViewCon
...
+			[APIClient checkinWithCode:resultAsString success:^(Checkin *checkin) {
+				self.checkinStatusView.hidden = NO;
+			} failure:nil];
+			NSLog(@"stop");
```
Comment (tsldh) [inline BlockHunt/CheckinViewController.m]: useless log

File: BlockHunt/Models/Checkin.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Cristina on 1/24/16.
+//  Copyright © 2016 SilverLogic. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import "AbstractModel.h"
+#import <CoreLocation/CoreLocation.h>
+
+@interface Checkin : AbstractModel
```
Comment (tsldh) [inline BlockHunt/Models/Checkin.h]: missing store field

File: BlockHunt/Models/Checkin.m [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#import "LocationHelper.h"
+
+@implementation Checkin
+
+
++ (NSDictionary*)fieldMappings {
+	// don't care so much about this
+	NSMutableDictionary *fieldMappings = [NSMutableDictionary dictionaryWithDictionary:[super fieldMappings]];
+	[fieldMappings addEntriesFromDictionary:@{
+											  @"id": @"checkinId",
```
Comment (tsldh) [inline BlockHunt/Models/Checkin.m:20]: missing store

File: BlockHunt/Singletons/APIClient.m [hunk 5]
```diff
@@ -190,6 +191,29 @@ + (void)getStoresAroundLocation:(CLLocationCoordinate2D)coordinate success:(void
     }];
 }
 
++ (void)checkinWithCode:(NSString *)qrcode success:(void (^)(Checkin *))success failure:(void (^)(NSError *, NSHTTPURLResponse *))failure {
+    NSDictionary *params = @{
+                             @"qrcode": qrcode,
+							 @"coords": @{
```
Comment (tsldh) [inline BlockHunt/Singletons/APIClient.m:198]: weird indentation

File: BlockHunt/Singletons/APIClient.m [hunk 6]
```diff
@@ -190,6 +191,29 @@ + (void)getStoresAroundLocation:(CLLocationCoordinate2D)coordinate success:(void
     }];
 }
 
++ (void)checkinWithCode:(NSString *)qrcode success:(void (^)(Checkin *))success failure:(void (^)(NSError *, NSHTTPURLResponse *))failure {
+    NSDictionary *params = @{
+                             @"qrcode": qrcode,
```
Comment (tsldh) [inline BlockHunt/Singletons/APIClient.m:197]: also pass @"expand": @"store"

**Pull request merged**